include/openenclave/internal/trace.h

```diff
@@ -1,53 +1,112 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-#ifndef _OE_TRACE_H
-#define _OE_TRACE_H
+#ifndef _OETRACE_H
+#define _OETRACE_H
 
 #include <openenclave/bits/defs.h>
 #include <openenclave/bits/result.h>
+#include <openenclave/bits/types.h>
+
+// logging flags
+#define OE_LOG_FLAGS_NONE 0x00000000
+#define OE_LOG_FLAGS_ATTESTATION 0x00000001
+#define OE_LOG_FLAGS_GET_REPORT 0x00000002
+#define OE_LOG_FLAGS_VERIFY_REPORT 0x00000004
+#define OE_LOG_FLAGS_COMMON 0x00000008
+#define OE_LOG_FLAGS_CERT 0x00000010
+#define OE_LOG_FLAGS_TOOLS 0x00000020
+#define OE_LOG_FLAGS_CRYPTO 0x00000040
+#define OE_LOG_FLAGS_SGX_SPECIFIC 0x00000100
+#define OE_LOG_FLAGS_IMAGE_LOADING 0x00000200
+#define OE_LOG_FLAGS_OTHERS 0x00000400
+#define OE_LOG_FLAGS_ALL 0xffffffff
+
+typedef enum _log_level_ {
+    OE_LOG_LEVEL_NONE = 0,
+    OE_LOG_LEVEL_FATAL,
+    OE_LOG_LEVEL_ERROR,
+    OE_LOG_LEVEL_WARNING,
```

Comment:
Thanks for the suggestion. 
FATAL was added to indicate critical condition has happened, different from normal API error, for example oe_abort was called and an enclave will no longer to function correctly.
WARNING can be use to spew out message like depreciated APIs still be used, it's not an error but it will in the future.
For TRACE, I was playing with a similar flag (DEBUG) for the same purpose : dumping out very detailed information. After some experiments, it's confusing when deciding whether to use INFO or DEBUG(TRACE).  With that, I will keep current levels to see how it goes for now, we can always 
new levels in the future.